packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -372,30 +380,65 @@ class ClearSelectionEvent extends SelectionEvent {
   const ClearSelectionEvent(): super._(SelectionEventType.clear);
 }
 
+/// An abstract base class for selection events that select a boundary at a location.
+///
+/// See also:
+/// * [SelectWordSelectionEvent], for an event that selects the word at the location.
+/// * [SelectParagraphSelectionEvent], for an event that selects the paragraph at the location.
+/// * [SelectLineSelectionEvent], for an event that selects the line at the location.
+/// * [SelectionEventType], for determining the subclass types.
+abstract class SelectBoundarySelectionEvent extends SelectionEvent {
+  const SelectBoundarySelectionEvent._(this.globalPosition, SelectionEventType type) : super._(type);
+
+  /// The position in global coordinates to select the boundary at.
+  final Offset globalPosition;
+}
+
+/// An abstract base class for selection events that select a boundary that
+/// may span multiple [Selectable]s.
+///
+/// See also:
+/// * [SelectParagraphSelectionEvent], for an event that selects the paragraph at the location.
+/// * [SelectLineSelectionEvent], for an event that selects the line at the location.
+/// * [SelectionEventType], for determining the subclass types.
+abstract class SelectMultiSelectableBoundarySelectionEvent extends SelectBoundarySelectionEvent {
+  const SelectMultiSelectableBoundarySelectionEvent._(this.absorb, Offset globalPosition, SelectionEventType type) : super._(globalPosition, type);
+
+  /// Whether the [Selectable] receiving the event should be completely
+  /// absorbed into a boundary.
```

Comment:
I can't parse this sentence either.

Reply:
it feel to me the `boundary` word has other meaning in this doc, can you explain a bit more?

Reply:
By boundary I meant a range within some content with a start and end. This boundary can be across multiple `Selectable`s. Some `Selectable`s within a boundary should be able to be considered as part of that boundary as a single unit rather than granularly. Let me know if I can explain more.